cep47/src/entrypoints.rs

```diff
@@ -0,0 +1,188 @@
+use std::collections::BTreeSet;
+
+use casper_contract::{
+    contract_api::{runtime, storage},
+    unwrap_or_revert::UnwrapOrRevert,
+};
+use casper_types::{
+    CLType, CLTyped, ContractPackageHash, EntryPoint, EntryPointAccess, EntryPointType,
+    EntryPoints, Key, Parameter,
+};
+use cep47_logic::{Meta, TokenId};
+
+pub fn get_entrypoints(package_hash: Option<ContractPackageHash>) -> EntryPoints {
+    let secure = if let Some(contract_package_hash) = package_hash {
+        let deployer_group = storage::create_contract_user_group(
+            contract_package_hash,
+            "deployer",
+            1,
+            BTreeSet::default(),
+        )
+        .unwrap_or_revert();
+        runtime::put_key("deployer_group_access", Key::URef(deployer_group[0]));
+        true
+    } else {
+        false
+    };
+
+    let mut entry_points = EntryPoints::new();
+    entry_points.add_entry_point(endpoint("name", vec![], CLType::String, None));
+    entry_points.add_entry_point(endpoint("symbol", vec![], CLType::String, None));
+    entry_points.add_entry_point(endpoint("meta", vec![], Meta::cl_type(), None));
+    entry_points.add_entry_point(endpoint("total_supply", vec![], CLType::U256, None));
+    entry_points.add_entry_point(endpoint("is_paused", vec![], CLType::Bool, None));
+    entry_points.add_entry_point(endpoint(
+        "pause",
+        vec![],
+        CLType::Unit,
+        if secure { Some("deployer") } else { None },
+    ));
+    entry_points.add_entry_point(endpoint(
+        "unpause",
+        vec![],
+        CLType::Unit,
+        if secure { Some("deployer") } else { None },
+    ));
+    entry_points.add_entry_point(endpoint(
+        "balance_of",
+        vec![Parameter::new("account", CLType::Key)],
+        CLType::U256,
+        None,
+    ));
+    entry_points.add_entry_point(endpoint(
+        "owner_of",
+        vec![Parameter::new("token_id", TokenId::cl_type())],
+        CLType::Option(Box::new(CLType::Key)),
+        None,
+    ));
+    entry_points.add_entry_point(endpoint(
+        "token_meta",
+        vec![Parameter::new("token_id", TokenId::cl_type())],
+        CLType::Option(Box::new(CLType::String)),
+        None,
+    ));
+    entry_points.add_entry_point(endpoint(
+        "tokens",
+        vec![Parameter::new("owner", CLType::Key)],
+        CLType::List(Box::new(CLType::String)),
+        None,
+    ));
+    entry_points.add_entry_point(endpoint(
+        "total_supply",
+        vec![Parameter::new("owner", CLType::Key)],
+        CLType::List(Box::new(CLType::String)),
+        None,
+    ));
+    entry_points.add_entry_point(endpoint(
+        "mint_one",
+        vec![
+            Parameter::new("recipient", CLType::Key),
+            Parameter::new("token_ids", CLType::Option(Box::new(TokenId::cl_type()))),
+            Parameter::new("token_meta", Meta::cl_type()),
+        ],
+        CLType::Unit,
+        if secure { Some("deployer") } else { None },
```

Comment:
It's more efficient and normal way to provide Mint/Burn availability to normal users rather than allowing only deployer_groups to do.

If a custom NFT contract needs to restrict the mint/burn function calls, they can rewrite the entry point access

Reply:
We can do that in the follow up PR.